tobac/utils/internal/xarray_utils.py

```diff
@@ -41,10 +173,252 @@ def find_vertical_axis_from_coord(
+    return_feat_df = copy.deepcopy(feature_df)
+
+    time_dim_name: str = "time"
+    # first, we must find the names of the dimensions corresponding to the numbered
+    # dimensions.
+
+    ndims: int = variable_da.ndim
+
+    time_dim_number = find_axis_from_dim(variable_da, time_dim_name)
+
+    is_3d = (time_dim_number is not None and ndims == 4) or (
+        time_dim_number is None and ndims == 3
+    )
+    vdim_coord = None
+    if is_3d:
+        hdim1_axis, hdim2_axis, vertical_axis = find_hdim_axes_3d(
+            variable_da,
+            vertical_coord,
+            vertical_axis,
+            time_dim_coord_name=time_dim_name,
+            return_vertical_axis=True,
+        )
+        if vertical_axis is None:
+            vdim_coord = find_vertical_coord_name(variable_da, vertical_coord)
+            vertical_axis = find_axis_from_dim_coord(variable_da, vdim_coord)
+        else:
+            vdim_coord = variable_da.dims[vertical_axis]
+    else:  # 2D
+        if ndims == 2:
+            hdim1_axis = 0
+            hdim2_axis = 1
+        elif ndims == 3 and time_dim_number is not None:
+            possible_dims = [0, 1, 2]
+            possible_dims.pop(time_dim_number)
+            hdim1_axis, hdim2_axis = possible_dims
+        else:
+            raise ValueError("DataArray has too many or too few dimensions")
+
+    # If the dimensions share a name with the coordinates and those coordinates do not match
+    # with the i, j, k-style indices, you cannot `interp` along those i, j, k indices.
+    # so, instead, we rename the dimensions to random strings so that we can
+    # run interpolation.
+
+    hdim1_name_original = variable_da.dims[hdim1_axis]
+    hdim2_name_original = variable_da.dims[hdim2_axis]
+
+    # generate random names for the new coordinates that are based on i, j, k values
+    hdim1_name_new = "".join(
+        random.choice(string.ascii_uppercase + string.ascii_lowercase + string.digits)
+        for _ in range(16)
+    )
+    hdim2_name_new = "".join(
+        random.choice(string.ascii_uppercase + string.ascii_lowercase + string.digits)
+        for _ in range(16)
+    )
+
+    dim_new_names = {
+        hdim1_name_original: hdim1_name_new,
+        hdim2_name_original: hdim2_name_new,
+    }
+    dim_interp_coords = {
+        hdim1_name_new: xr.DataArray(return_feat_df["hdim_1"].values, dims="features"),
+        hdim2_name_new: xr.DataArray(return_feat_df["hdim_2"].values, dims="features"),
+    }
+
+    if is_3d:
+        vdim_name_original = variable_da.dims[vertical_axis]
+        vdim_name_new = "".join(
+            random.choice(
+                string.ascii_uppercase + string.ascii_lowercase + string.digits
+            )
+            for _ in range(16)
+        )
+        dim_interp_coords[vdim_name_new] = xr.DataArray(
+            return_feat_df["vdim"].values, dims="features"
+        )
+
+        dim_new_names[vdim_name_original] = vdim_name_new
```

Comment:
I'm not a fan of using randomly generated names here. I would rather use a set of fixed names, similar to what we use for renaming dataframe columns for predicitive tracking. e.g. `__temp_hdim1_name`, `__temp_vdim_name`

Reply:
Yeah, adding in the random names is not a great idea. I've taken your suggestion here and added a check to see if these names are otherwise used, these are reasonable barriers to add.